### PR TITLE
add admin part

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: NewsletterAdminBundle/Controller/Admin/NewsletterSubscriberController.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        $form = $this->createForm('newsletter_subscriber', $newsletterSubscriber, array(
+            'attr' => array('class' => 'new'),
+            'action' => $this->generateUrl('open_orchestra_newsletter_subscriber_form', array('newsletterSubscriberId' => $newsletterSubscriberId)),
+            'method' => 'POST',
+        ));
+
+        $form->handleRequest($request);
+        $message = $this->get('translator')->trans('open_orchestra_newsletter.form.newsletter_subscriber.edit.success');
+
+        if ($this->handleForm($form, $message, $newsletterSubscriber)) {
```
Comment (alavieille) [inline NewsletterAdminBundle/Controller/Admin/NewsletterSubscriberController.php]: ??

File: NewsletterAdminBundle/NavigationPanel/NewsletterSubscriberPanelStrategy.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace OpenOrchestra\NewsletterAdminBundle\NavigationPanel;
+
+use OpenOrchestra\Backoffice\NavigationPanel\Strategies\AbstractNavigationPanelStrategy;
+
+/**
+ * Class NewsletterSubscriberPanelStrategy
+ */
+class NewsletterSubscriberPanelStrategy extends AbstractNavigationPanelStrategy
```
Comment (alavieille) [inline NewsletterAdminBundle/NavigationPanel/NewsletterSubscriberPanelStrategy.php:10]: can you redefine the method `getWeight`

File: NewsletterAdminBundle/Resources/translations/messages.fr.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    left_menu:
+        administration:
+            newsletter_subscriber:
+                list: Inscris à la newsletter
+    form:
+        newsletter_subscriber:
+            edit:
+                success: L'inscris a été mis à jour
+            new:
+                success: L'inscris a été crée
```
Comment (alavieille) [inline NewsletterAdminBundle/Resources/translations/messages.fr.yml]: créé

File: NewsletterAdminBundle/Resources/views/NavigationPanel/newsletter_subscriber.html.twig [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<a
+        href="#newsletter_subscriber/list"
+        id="nav-newsletter_subscriber"
+        data-url="{{ path('open_orchestra_api_newsletter_subscriber_list') }}"
+        data-add="{{ path('open_orchestra_newsletter_subscriber_new') }}"
+        data-displayed-elements="last_name, first_name, email, site_id"
+        data-visible-elements="last_name, first_name, email, site_id"
+        data-translated-header="last_name, first_name, email, site_id"
```
Comment (alavieille) [inline NewsletterAdminBundle/Resources/views/NavigationPanel/newsletter_subscriber.html.twig]: why not translate elements header ?

File: NewsletterAdminBundle/Transformer/NewsletterSubscriberTransformer.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ * Class NewsletterSubscriberTransformer
+ */
+class NewsletterSubscriberTransformer extends AbstractTransformer
+{
+    /**
+     * @param mixed|NewsletterSubscriberInterface $newsletterSubscriber
+     *
+     * @return FacadeInterface
+     */
+    public function transform($newsletterSubscriber)
```
Comment (alavieille) [inline NewsletterAdminBundle/Transformer/NewsletterSubscriberTransformer.php:22]: can you check if $newsletterSubscriber is an instance of NewsletterSubscriberInterface and throw an `TransformerParameterTypeException()` if it isn't like other transformer.

File: NewsletterAdminBundle/Controller/Admin/NewsletterSubscriberController.php (hunk 1 above)
Comment (nicolasThal) [inline NewsletterAdminBundle/Controller/Admin/NewsletterSubscriberController.php]: At first I wanted to do something inside, but I think I will suppress the if

File: NewsletterAdminBundle/Resources/views/NavigationPanel/newsletter_subscriber.html.twig (hunk 4 above)
Comment (nicolasThal) [inline NewsletterAdminBundle/Resources/views/NavigationPanel/newsletter_subscriber.html.twig]: I was lazy, I will do it to finalise the PR

**Comment by nicolasThal** [comment]
@itkg-alavieille pr updated

File: NewsletterAdminBundle/DependencyInjection/Configuration.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace OpenOrchestra\NewsletterAdminBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+/**
+ * This is the class that validates and merges configuration from your app/config files
+ *
+ * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html#cookbook-bundles-extension-config-class}
```
Comment (itkg-ngilain) [inline NewsletterAdminBundle/DependencyInjection/Configuration.php]: Are those comments usefull ?

File: NewsletterAdminBundle/DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ */
+class Configuration implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
```
Comment (itkg-ngilain) [inline NewsletterAdminBundle/DependencyInjection/Configuration.php]: These one should be more detailed I think

File: NewsletterAdminBundle/Resources/translations/messages.fr.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+open_orchestra_newsletter:
+    left_menu:
+        administration:
+            newsletter_subscriber:
+                list: Inscris à la newsletter
```
Comment (ngilain) [inline NewsletterAdminBundle/Resources/translations/messages.fr.yml]: Inscrits

I know that was me who named that, but with a second look at this, I think we should translate by "Abonnés"

File: NewsletterAdminBundle/Resources/translations/messages.fr.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+            element:
+                last_name: Nom
+                first_name: Prénom
+                email: Email
+                site_id: Site
+    form:
+        newsletter_subscriber:
+            edit:
+                success: L'inscris a été mis à jour
```
Comment (ngilain) [inline NewsletterAdminBundle/Resources/translations/messages.fr.yml]: L'inscrit / L'abonné

File: NewsletterBundle/Resources/translations/messages.fr.yml [hunk 10]
```diff
@@ -1,3 +1,9 @@
 open_orchestra_newsletter:
     title:
         registration: Inscription à la lettre d'information
+    registration:
+        success: Vous etes inscrit à la newsletter
```
Comment (ngilain) [inline NewsletterBundle/Resources/translations/messages.fr.yml]: êtes

**Comment by itkg-rboessel** [comment]
:+1: with removing usefulness comment  and change weight

File: NewsletterAdminBundle/DependencyInjection/OpenOrchestraNewsletterAdminExtension.php [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * This is the class that loads and manages your bundle configuration
+ *
+ * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html}
+ */
+class OpenOrchestraNewsletterAdminExtension extends Extension
+{
+    /**
+     * {@inheritdoc}
```
Comment (itkg-rboessel) [inline NewsletterAdminBundle/DependencyInjection/OpenOrchestraNewsletterAdminExtension.php]: real comment

File: NewsletterAdminBundle/NavigationPanel/NewsletterSubscriberPanelStrategy.php (hunk 2 above)
Comment (nicolasThal) [inline NewsletterAdminBundle/NavigationPanel/NewsletterSubscriberPanelStrategy.php:10]: done

**Comment by nicolasThal** [comment]
@itkg-rboessel @itkg-ngilain pr updated

**Pull request merged**